lib/plugins/aws/package/compile/functions.js

```diff
@@ -421,31 +445,41 @@ class AwsCompileFunctions {
       });
 
       for (const layerArtifactPath of layerArtifactPaths.sort()) {
-        await addFileContentsToHashes(layerArtifactPath, [versionHash]);
+        await this.addFileToHash(layerArtifactPath, versionHash);
       }
 
       // Include function and layer configuration details in the version id hash
       for (const layerConfig of layerConfigurations) {
         delete layerConfig.properties.Content.S3Key;
       }
 
-      // sort the layer conifigurations for hash consistency
-      const sortedLayerConfigurations = {};
-      const byKey = ([key1], [key2]) => key1.localeCompare(key2);
-      for (const { name, properties: layerProperties } of layerConfigurations) {
-        sortedLayerConfigurations[name] = _.fromPairs(Object.entries(layerProperties).sort(byKey));
-      }
-
       const functionProperties = _.cloneDeep(functionResource.Properties);
       // In `image` case, we assume it's path to ECR image digest
       if (!functionObject.image) delete functionProperties.Code;
       // Properties applied to function globally (not specific to version or alias)
       delete functionProperties.ReservedConcurrentExecutions;
       delete functionProperties.Tags;
-      functionProperties.layerConfigurations = sortedLayerConfigurations;
 
-      const sortedFunctionProperties = _.fromPairs(Object.entries(functionProperties).sort(byKey));
-      versionHash.write(JSON.stringify(sortedFunctionProperties));
+      const lambdaHashingVersion = this.serverless.service.provider.lambdaHashingVersion;
+      if (lambdaHashingVersion === '20201221') {
```

Comment:
Let's do `if (lambdaHashingVersion)`

Reply:
Thanks for catching, I missed that line :+1: